services/provider/proto/provider.proto

```diff
@@ -109,3 +122,50 @@ message AcknowledgeOnboardingRequest{
 message AcknowledgeOnboardingResponse{
 
 }
+
+// ProvisionStorageClassClaimRequest holds the information required to
+// create the StorageclassClaim CR on provider cluster.
+message ProvisionStorageClassClaimRequest{
+    // K8s UID (UUID) of the consumer cluster
+    string storageConsumerUUID = 1;
+    // storageclassClaim (UUID) of the consumer cluster.
+    string storageClassClaimUUID = 2;
+    // StorageType of the storageclassClaim
+    enum StorageType {
+      CephFS = 0;
+      RBD = 1;
+   }
+   StorageType storageType = 3;
+}
```

Comment:
we need a field for encryption as well.

Reply:
encryption is part of the storageclass we can handle it on the consumer side as well IMO no need to make it part of the RPC call. WDYT?

Reply:
So once we get back StorageClass object in response, we inject encryption spec and create it?
Sounds good.

Reply:
I am not sure this is a good approach, I think we want the ability to understand the exact claim configuration on the provider as well, even if we are not going to take action on it. The main reason here is trackability and supportability

Reply:
Yes but this is extra data on the RPC call in both request and the response. if that is fine am fine with adding a field for the encryption as well.